File: codewars/11-revers_words/golang/main.go
package main

import (
	"fmt"
	"strings"
)

func reverse(str string) (result string) {
	for _, v := range str {
		result = string(v) + result
	}
	return
}

func ReverseWords(str string) string {
	ans := ""
	arr := strings.Split(str, " ")
	for _, v := range arr {
		ans += reverse(v)
		ans += " "
	}
	ans = strings.TrimSuffix(ans, " ")
	return ans
}

// Another solution without imports
// AWESOME!!!
func ReverseWords2(str string) string {
	var rev string
	var word string

	for _, i := range str {
		if i == ' ' {
			rev = rev + word + " " // Adds word and space to result
			word = ""              // Empties word variable
		} else {
			word = string(i) + word // Adds letter to temporary word variable
		}
	}

	return rev + word // reverse those words
}

func main() {
	fmt.Println(ReverseWords("Hello  World ") + "|")
}

File: codewars/11-revers_words/python/main.py
def reverse_words(text):
    text = text.split(" ")
    ans = ""
    for v in text:
        ans += v[::-1] + " "
    return ans[:-1]

# Another solutions
def reverse_words2(str):
    return ' '.join(s[::-1] for s in str.split(' '))

def reverse_words3(str):
  newStr = []
  for i in str.split(' '):
      newStr.append(i[::-1])
  return ' '.join(newStr)

import re

def reverse_words4(str):
  return re.sub(r'\S+', lambda w: w.group(0)[::-1], str)

if __name__ == "__main__":
    print(reverse_words("Hello  World"))